app/presenters/nfg_onboarder/high_level_navigation_bar_presenter.rb

```diff
@@ -20,25 +22,45 @@ def render_previous_button_unless?
     # for the step nav body text
     # or default to humanizing the
     # step's symbol.
-    def step_body(step)
-      I18n.t(step, scope: h.locale_namespace + [:step_navigations], default: step.to_s.humanize)
+    def step_body(nav_step)
+      I18n.t(
+        nav_step,
+        scope: h.locale_namespace + [:step_navigations],
+        default: nav_step.to_s.split('_').map(&:capitalize).join(' ')
+      )
     end
 
     # Returns a check mark for the last step's icon
-    def step_icon(step)
-      step == all_steps.last ? 'check' : nil
+    def step_icon(nav_step)
+      nav_step == all_steps.last ? 'check' : nil
     end
 
     # Returns an nfg_ui friendly status for the :step component's status trait
     #
     # Example usage:
     # = ui.nfg :step, onboarder_presenter.step_status(the_step), step: 1, href: wizard_path(the_step)
-    def step_status(step)
-      return :active if step.to_sym == active_step.to_sym
-      return :visited if try(:completed_steps, h.controller_name).try(:include?, step)
-      # in case steps change, if the step or active step can't be found in what
-      # is currently the list of steps  don't barf
-      return :disabled if (all_steps.index(step.to_sym) || 0) > (all_steps.index(active_step.to_sym) || 0)
+    def step_status(nav_step)
+      return :active if active?(nav_step)
+      return :visited if visited?(nav_step)
+      return :disabled if disabled?(nav_step)
+    end
+
+    def active?(nav_step)
+      nav_step.to_sym == h.controller_name.to_sym
+    end
+
+    def disabled?(nav_step)
+      (all_steps.index(nav_step.to_sym) || 0) > (all_steps.index(active_step.to_sym) || 0)
+    end
+
+    def visited?(nav_step)
+      model.onboarder_progress.keys.include?(nav_step.to_s)
```

Comment:
I think the previous idea was that we need to keep track of each controller's progress. Does this still do that?